clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -1206,10 +1206,21 @@ public void testUncaughtExceptionInHeartbeatThread() throws Exception {
                 throw e;
             return false;
         }, heartbeatResponse(Errors.UNKNOWN_SERVER_ERROR));
+        coordinator.ensureActiveGroup();
+        mockTime.sleep(HEARTBEAT_INTERVAL_MS);
+
+        try {
```

Comment:
Do you think It could be cleaner to use assertThrows? Something like. In this case we won't need to fail then assertEquals. (And the test message can be more accurate)
```
assertThrows(SadException.class,
        () -> coordinator.sadPath());
```

Reply:
I agree with you. But here, if the method coordinator.timeToNextHeartbeat(0) or coordinator.pollHeartbeat(mockTime.milliseconds()) is called **only once**, an exception may not be thrown.
We expect that within one second, the background heartbeat thread can run and eventually exit abnormally.